src/gpu/utils.rs

```diff
@@ -62,3 +64,20 @@ pub fn get_core_count(d: Device) -> GPUResult<usize> {
         }),
     }
 }
+
+#[derive(Debug)]
+pub struct LockedFile(File);
+
+pub const LOCK_NAME: &str = "/tmp/bellman.lock";
+
+pub fn lock() -> io::Result<LockedFile> {
+    let file = File::create(LOCK_NAME)?;
```

Comment:
One more request, lets write an `info` log when locking and unlocking, just so we have a chance to debug this if it doesn't work as expected.